### PR TITLE
Coverage/patch contains excessive properties error

### PR DESCRIPTION
**Pull request by NemanjaStefanovic**
@nickevansuk Can you just check documentation should type be PatchContainsExcessiveProperties or PatchContainsExcessivePropertiesError (https://openactive.io/open-booking-api/EditorsDraft/#orderproposal-update) ? 

Issue #218

### CODE REVIEW

**Comment by nickevansuk** [comment]
Yes, good catch! The spec is inconsistent here. Will raise some issues to resolve this, and then get back to addressing these PRs.

File: packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+const { expect } = chakram;
+/* eslint-enable no-unused-vars */
+
+FeatureHelper.describeFeature(module, {
+  testCategory: 'cancellation',
+  testFeature: 'customer-requested-cancellation',
+  testFeatureImplemented: true,
+  testIdentifier: 'patch-contains-excessive-properties-error',
+  testName: 'Successful booking and unsuccessful cancellation due to PatchContainsExcessiveProperties error',
```
Comment (nickevansuk) [inline packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js:16]:
```suggestion
  testName: 'Successful booking and unsuccessful cancellation due to PatchContainsExcessivePropertiesError',
```

File: packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    });
+
+    it('Order Cancellation return 400 on fail', function () {
+      expect(state.uResponse).to.have.status(400);
+    });
+
+    it('Order Cancellation should have PatchContainsExcessiveProperties as @type and expected description', function () {
+      expect(state.uResponse.body['@type']).to.equal('PatchContainsExcessivePropertiesError');
+      expect(state.uResponse.body.description).to.equal('PATCH includes unexpected properties that are not permitted.');
+    });
```
Comment (nickevansuk) [inline packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js:88]: Note we should not be checking this description is equal to something specific. This should use https://github.com/openactive/openactive-test-suite/blob/f189f5367f7fc680597c8ae5ea224fb194793781/packages/openactive-integration-tests/test/shared-behaviours/errors.js#L14

File: packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  describe('Orders Feed', function () {
+    beforeAll(async function () {
+      await flow.U();
+    });
+
+    it('Order Cancellation return 400 on fail', function () {
+      expect(state.uResponse).to.have.status(400);
+    });
+
+    it('Order Cancellation should have PatchContainsExcessiveProperties as @type and expected description', function () {
```
Comment (nickevansuk) [inline packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js:85]:
```suggestion
    it('Order Cancellation should have PatchContainsExcessivePropertiesError as @type and expected description', function () {
```

**Review by nickevansuk — CHANGES_REQUESTED** [review]
Spec now updated to `PatchContainsExcessivePropertiesError`

File: packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js (hunk 2 above)
Comment (NemanjaStefanovic) [reply packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js]: Ok done

File: packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+FeatureHelper.describeFeature(module, {
+  testCategory: 'cancellation',
+  testFeature: 'customer-requested-cancellation',
+  testFeatureImplemented: true,
+  testIdentifier: 'patch-contains-excessive-properties-error',
+  testName: 'Successful booking and unsuccessful cancellation due to PatchContainsExcessivePropertiesError',
+  testDescription: 'PatchContainsExcessivePropertiesError returned because patch request includes other properties than @type, @context, orderProposalStatus and orderCustomerNote',
+  // The primary opportunity criteria to use for the primary OrderItem under test
+  testOpportunityCriteria: 'TestOpportunityBookableCancellable',
```
Comment (nickevansuk) [inline packages/openactive-integration-tests/test/features/cancellation/customer-requested-cancellation/implemented/patch-contains-excessive-properties-error-test.js:20]: This requires a spec update, as it is not possible to filter for `TestOpportunityBookableCancellable` right now. Approving for now and noting this as a separate issue.

**Review by nickevansuk — APPROVED** [review]

**Pull request merged**